### PR TITLE
Add Benzene and Mercurius server performance tests

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

> **Note**
> 
> Please provide a description of the work completed in this PR below

Adds Mercurius and Benzene server performance tests - initial observations are that Mercurius (Fastify) is significantly better for low complexity queries. 

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High

Complexity: Low

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 60 | 52 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 102 | 140 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 64 | 77 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3965 | 4057 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 3555 | 3701 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 52 | 68 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 51 | 65 | 2174 |
|  aggregations.DeeplyNestedCount | 13070331 | 13070331 | 5784 | 5802 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 171 | 208 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 68 | 89 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 198 | 310 | 3003 |
|  connections.Connection | 14082 | 14082 | 63 | 88 | 2174 |
|  connections.NestedConnection | 45466 | 45472 | 136 | 182 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 75 | 88 | 1040 |
|  connections.ConnectionWithSortAndCypher | 3284 | 3284 | 295 | 451 | 1040 |
|  create.SimpleMutation | 7 | 7 | 67 | 77 | 1 |
|  cypher-directive.TopLevelCypherDirective | 122100 | 122100 | 673 | 837 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 153619 | 153617 | 203 | 291 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 374 | 328 | 1040 |
|  delete.NestedDeleteInUpdate | 23202 | 23202 | 204 | 348 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 65 | 60 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 125 | 143 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 46 | 54 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 70 | 89 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 28 | 35 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 53 | 63 | 2174 |
|  query.QueryWhere | 9698 | 9699 | 51 | 52 | 2161 |
|  query.SimpleQueryWithNestedWhere | 9876 | 9877 | 67 | 80 | 2161 |
|  query.Nested | 10096041 | 10096041 | 12618 | 12452 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 12568 | 12016 | 2004000 |
|  query.OrFilterOnRelationships | 44314 | 41010 | 276 | 315 | 2167 |
|  query.OrFilterOnRelationshipsAndNested | 38183 | 34884 | 316 | 377 | 2167 |
|  query.QueryWithNestedIn | 13425 | 13212 | 69 | 116 | 1223 |
|  query.NestedConnectionWhere | 9834 | 9834 | 68 | 92 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9911 | 9909 | 99 | 124 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19134 | 19086 | 184 | 211 | 1598 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 223 | 318 | 1134 |
|  query.Fulltext | 80 | 80 | 46 | 57 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 57 | 65 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 90 | 117 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1414 | 1436 | 292 | 409 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 62 | 77 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 109 | 119 | 4516 |
|  sorting.SortWithTopLevelCypher | 3121 | 3121 | 207 | 321 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 91 | 83 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 133 | 126 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 418 | 459 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 366 | 368 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 135 | 189 | 2002 |

</details>
        
Old Schema Generation: 57.889s
Schema Generation: 58.739s
Old Subgraph Schema Generation: 1:12.699 (m:ss.mmm)
Subgraph Schema Generation: 1:18.146 (m:ss.mmm)

**Comment by angrykoala** [comment]
Could be nice to mention in the comment the comparison that you found :)

File: packages/graphql/tests/performance/server/package.json [hunk 1]
```diff
@@ -9,14 +9,21 @@
     "scripts": {
         "doctor-apollo": "clinic doctor --on-port -- node ./servers/apollo.js",
         "flame-apollo": "clinic flame -- node ./servers/apollo.js",
+        "doctor-benzene": "clinic doctor --on-port -- node ./servers/benzene.js",
+        "flame-benzene": "clinic flame -- node ./servers/benzene.js",
+        "doctor-mercurius": "clinic doctor --on-port -- node ./servers/mercurius.js",
+        "flame-mercurius": "clinic flame -- node ./servers/mercurius.js",
```
Comment (angrykoala) [inline packages/graphql/tests/performance/server/package.json:15]: We need to think of a better approach for this soon

File: packages/graphql/tests/performance/server/servers/benzene.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    });
+
+    server.listen(4000, () => {
+        console.info("Server is running on http://localhost:4000/graphql");
+    });
+}
+
+main().catch((err) => {
+    console.error(err);
+});
```
Comment (angrykoala) [inline packages/graphql/tests/performance/server/servers/benzene.js:78]: No trailing newline?

You monster

**Review by angrykoala — APPROVED** [review]

File: packages/graphql/tests/performance/server/servers/benzene.js (hunk 2 above)
Comment (darrellwarde) [reply packages/graphql/tests/performance/server/servers/benzene.js:78]:
```suggestion
});

```

File: packages/graphql/tests/performance/server/servers/mercurius.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    });
+
+    await app.listen({ port: 4000 }).then(() => {
+        console.info("Server is running on http://localhost:4000/graphql");
+    });
+}
+
+main().catch((err) => {
+    console.error(err);
+});
```
Comment (darrellwarde) [inline packages/graphql/tests/performance/server/servers/mercurius.js:60]:
```suggestion
});

```

**Review by angrykoala — DISMISSED** [review]

**Pull request merged**